webapi/src/main/scala/org/knora/webapi/messages/v2/responder/standoffmessages/StandoffMessagesV2.scala

```diff
@@ -0,0 +1,112 @@
+
+/**
+  * Represents a request to create a mapping between XML elements and attributes and standoff classes and properties.
+  * A successful response will be a [[CreateMappingResponseV2]].
+  *
+  * @param metadata the metadata describing the mapping.
+  * @param xml the mapping in XML syntax.
+  * @param userProfile the profile of the user making the request.
+  * @param apiRequestID the ID of the API request.
+  */
+case class CreateMappingRequestV2(metadata: CreateMappingRequestMetadataV2, xml: CreateMappingRequestXMLV2, userProfile: UserProfileV1, apiRequestID: UUID) extends StandoffResponderRequestV2
+
+/**
+  * Represents the metadata describing the mapping that is to be created.
+  *
+  * @param label the label describing the mapping.
+  * @param projectIri  the IRI of the project the mapping belongs to.
+  * @param mappingName the name of the mapping to be created.
+  */
+case class CreateMappingRequestMetadataV2(label: String, projectIri: SmartIri, mappingName: String) extends StandoffResponderRequestV2
+
+object CreateMappingRequestMetadataV2 extends KnoraJsonLDRequestReaderV2[CreateMappingRequestMetadataV2] {
+    override def fromJsonLD(jsonLDDocument: JsonLDDocument,
+                            apiRequestID: UUID,
+                            userProfile: UserProfileV1): CreateMappingRequestMetadataV2 = {
+
+        implicit val stringFormatter: StringFormatter = StringFormatter.getGeneralInstance
+
+        val label: String = jsonLDDocument.requireString(OntologyConstants.Rdfs.Label, stringFormatter.toSparqlEncodedString)
+
+        val projectIri: SmartIri = jsonLDDocument.requireString(OntologyConstants.KnoraApiV2WithValueObjects.AttachedToProject, stringFormatter.toSmartIriWithErr)
+
+        val mappingName: String = jsonLDDocument.requireString(OntologyConstants.KnoraApiV2WithValueObjects.MappingHasName, stringFormatter.toSparqlEncodedString)
+
+        CreateMappingRequestMetadataV2(
+            label = label,
+            projectIri = projectIri,
+            mappingName = mappingName
+        )
+    }
+}
+
+/**
+  * Represents the mapping as an XML document.
+  *
+  * @param xml the mapping to be created.
+  */
+case class CreateMappingRequestXMLV2(xml: String) extends StandoffResponderRequestV2
+
+/**
+  * Provides the IRI of the created mapping.
+  *
+  * @param mappingIri the IRI of the resource (knora-base:XMLToStandoffMapping) representing the mapping that has been created.
+  */
+case class CreateMappingResponseV2(mappingIri: IRI) extends KnoraResponseV2 {
+
+    def toJsonLDDocument(targetSchema: ApiV2Schema, settings: SettingsImpl): JsonLDDocument = {
+
+        implicit val stringFormatter: StringFormatter = StringFormatter.getGeneralInstance
+
+        val body = JsonLDObject(Map(
+            "@id" -> JsonLDString(mappingIri),
+            "@type" -> JsonLDString(OntologyConstants.KnoraBase.XMLToStandoffMapping.toSmartIri.toOntologySchema(targetSchema).toString)
```

Comment:
@benjamingeer Is this the correct way to use smartIri (following <http://www.knora.org/documentation/manual/rst/03-knora-api-server/design-documentation/api-v2.html?highlight=smartiri#usage>)?

Is calling `toString` on it correct after having converted it to a target schema?

Reply:
Yes, `toString` gets you the IRI as a string, which is what you need for `JsonLDString`.

Converting to the target schema and then to JSON-LD in the same method is OK if you only need to support the complex schema, which I guess is the case here. If you need to be able to return JSON-LD in more than one schema, I recommend doing the schema conversion in a separate method.

Reply:
> Converting to the target schema and then to JSON-LD in the same method is OK if you only need to support the complex schema, which I guess is the case here. 

This is the response to a mapping creation request. Write or update operations will always be in v2 complex schema and thus v2 complex will be returned. But still the `targetSchema` is not hard-coded, it is a parameter. So it should theoretically also work for v2 simple, right?